src/persist-client/src/rpc.rs

```diff
@@ -214,7 +214,12 @@ impl GrpcPubSubClient {
         loop {
             metrics.pubsub_client.grpc_connection.connected.set(0);
 
-            if !PUBSUB_CLIENT_ENABLED.get(&config.persist_cfg) {
+            if !config
```

Comment:
I'm totally fine with the sleep, but if you wanted to do something slightly fancier (`tokio::sync::OnceCell<()>`?) to get back the notification you had in the other PR happy to accept that as well

Reply:
https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html or a oneshot channel are probably the easiest!

Reply:
Yeah, I tried a `OnceCell` but it doesn’t have the right API as far as I can tell. You can’t await someone _else_ setting the value in the cell.

A one shot channel with a shared receiver and a notify would work but require managing two separate fields, which seemed like more trouble than it was worth.

Reply:
> Yeah, I tried a OnceCell but it doesn’t have the right API as far as I can tell. You can’t await someone else setting the value in the cell.

Oh no kidding! I thought I'd checked this when we discussed this way back when, but indeed there is not. Bummer

Okay, may as well leave this as a sleep!

Reply:
New idea! You could store a `tokio::sync::watch::Sender<()>`, which can hand you a `Receiver` from fn [subscribe](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.subscribe). Voila, one field :)

Reply:
I think it has to be a `Arc<watch::Sender<bool>>`, but it came out really cleanly. Thank you for the suggestion!

Reply:
I always forget about `tokio::sync::watch`, which happens to be extremely useful